### DISCUSSION TITLE
How to delete a user from Authentication

### DISCUSSION THREAD

**Discussion by nappalm** [supabase/supabase#21274] [Questions]
I want to give users the ability to delete their user account, creating a cascade effect in the different tables where that authentication id is related.

Try to use the method:
```
export async function deleteAccount(uuid: string) {
  return await supabase.auth.admin.deleteUser(uuid);
}
```
but I get a 401 error even though the user's session is active.

Any solution?

**Comment by GaryAustin1** [comment]
Admin auth functions require they be run on a server or edge function.  You need to use the service_role key to execute them.
One example from a SB employee:  https://blog.mansueli.com/supabase-user-self-deletion-empower-users-with-edge-functions
